p2p/swarmhandlers.go

```diff
@@ -214,6 +219,9 @@ func (s *swarmImpl) onDisconnectionRequest(req node.RemoteNodeData) {
 func (s *swarmImpl) onSendHandshakeMessage(r SendMessageReq) {
 
 	// check for existing remote node and session
+	defer s.peerMapMutex.Unlock()
```

Comment:
defer is cool but it reported to have a performance penalty, so if we can do it without defer its better.